server/priv/marketing/blog/2025/09/22/scale.md

```diff
@@ -0,0 +1,93 @@
+
+The second path that we see some companies taking is [Bazel](https://bazel.build/). Bazel is a very advanced and powerful build system. But replacing an entire build system is even costlier than introducing React Native. Most programming languages and ecosystems come with some build system built-in: [Elixir](https://elixir-lang.org/), [Swift](https://www.swift.org/), Xcode, [Gradle](https://gradle.org/), [Cargo](https://doc.rust-lang.org/cargo/)... So when you replace it with something else, reconciling the build system piece with everything that builds upon it is a huge investment that not many companies are willing to make. At Tuist, we often see companies whose story goes like this: One or a few engineers get excited about Bazel, they convince leadership to spend a few months of work introducing Bazel, they manage to get it working although no one knows how it works except them, those engineers leave the company, and the company decides to drop Bazel because it's too costly. We've seen countless companies going through this path, and I completely get it. As much as you can, you should stay close to your build system because that also means staying close to the ecosystem that surrounds it.
+
+For those who prefer to stay closer to the native toolchain and therefore have a lower cost to scale up their development, they see Tuist as the best option. This is by design. **We abstracted just where we had to, such that we helped teams overcome the challenges while keeping an eye on and peeling abstractions when the issues were addressed at the right layer, like we saw with Xcode 16 and the introduction of [buildable folders](https://dimillian.medium.com/why-you-should-use-xcode-16-buildable-folders-instead-of-groups-6f438611914d).**
+
+At this point, you might wonder: what are those challenges that large companies are facing, and that I might face too, if not today, maybe tomorrow? Let's explore some of them.
+
+## 1. Derived Data
```

Comment:
I would maybe start with the more important problems, like build times, and leave derived data as a later point.